src/mmw/js/src/modeling/models.js

```diff
@@ -53,48 +67,71 @@ var ProjectModel = Backbone.Model.extend({
         this.set('name', newName);
     },
 
-    saveAll: function() {
-        var self = this;
-
-        // If we haven't saved the project before,
-        // hold on to the scenarios so that we can
-        // save each one once we have an ID for the project
-        if (this.isNew) {
-            var tempScenarios = this.get('scenarios');
+    saveProjectAndScenarios: function() {
+        if (App.user.get('uid') !== this.get('uid')) {
+            return;
         }
 
-        this.save()
-            .done(function(resp) {
-                console.log('Saved project');
-
-                var scenarios = tempScenarios ? tempScenarios : self.get('scenarios');
-
-                scenarios.each(function(scenario) {
-                    if (!scenario.get('project')) {
+        var self = this,
+            scenarios = this.get('scenarios');
+        if (!this.get('id')) {
+            // We haven't saved the project before, save the project and then
+            // set the project ID on each scenario, then reattach the senarios
+            // to the project.
+            this.save()
+                .done(function() {
+                    scenarios.each(function(scenario) {
                         scenario.set('project', self.get('id'));
-                    }
-
-                    scenario
-                        .save()
-                        .done(function(resp) {
-                            console.log('Saved scenario');
-                        })
-                        .fail(function(resp) {
-                            console.log('Failed to save scenario');
-                        });
+                    });
+                })
+                .fail(function() {
+                    console.log('Failed to save project');
+                });
+        } else {
+            this.save()
+                .fail(function() {
+                    console.log('Failed to save project');
                 });
-            })
-            .fail(function(resp) {
-                console.log('Failed to save project');
+        }
+    },
+
+    addIdsToScenarios: function() {
+        if (!this.get('id')) {
+            this.saveProjectAndScenarios();
```

Comment:
Both `addIdsToScenarios` and `saveProjectAndScenarios` can add Ids and save the project and scenarios.  Consider consolidating the methods or splitting them apart so they each are responsible for one or the other, as their names suggest.